core/core_test.go

```diff
@@ -75,6 +82,190 @@ func TestTranscode(t *testing.T) {
 	// TODO check transcode loop expiry, storage, sig construction, etc
 }
 
+func TestTranscodeLoop_GivenNoSegmentsPastTimeout_CleansSegmentChan(t *testing.T) {
+	//Set up the node
+	drivers.NodeStorage = drivers.NewMemoryDriver("")
+	db, _ := common.InitDB("file:TestTranscode?mode=memory&cache=shared")
+	defer db.Close()
+	seth := &eth.StubClient{}
+	tmp, _ := ioutil.TempDir("", "")
+	n, _ := NewLivepeerNode(seth, tmp, db)
+	defer os.RemoveAll(tmp)
+	ffmpeg.InitFFmpeg()
+	ss := StubSegment()
+	md := &SegTranscodingMetadata{Profiles: videoProfiles}
+	n.Transcoder = NewLocalTranscoder(tmp)
+
+	transcodeLoopTimeout = 100 * time.Millisecond
+	assert := assert.New(t)
+	require := require.New(t)
+
+	_, err := n.sendToTranscodeLoop(md, ss)
+	require.Nil(err)
+	segChan := getSegChan(n, md.ManifestID)
+	require.NotNil(segChan)
+
+	waitForTranscoderLoopTimeout(n, md.ManifestID)
+
+	segChan = getSegChan(n, md.ManifestID)
+	assert.Nil(segChan)
+}
+
+func TestTranscodeLoop_GivenOnePMSession_RedeemsOneSession(t *testing.T) {
```

Comment:
Should we include `_GivenNoSegmentsPastTimeout_` to the test function name?

Same question applies to the rest of the test function names.

Reply:
Maybe change the middle piece to say `GivenOnePMSessionAtVideoSessionTimeout` ?

Reply:
SGTM as long as the test function name conveys that we're specifically testing for the scenario after the transcode loop times out (since we already have a test function named `TestTranscodeLoop_GivenNoSegmentsPastTimeout_CleansSegmentChan` explicitly indicating we're testing after the transcode loop times out)